src/pages/admin/donate/components/generic-form/GenericForm.tsx

```diff
@@ -0,0 +1,397 @@
+                                                    handleBlur={handleBlur(f.name)}
+                                                    onlyNumbers={f.onlyNumbers}
+                                                    maxLength={f.maxLength}
+                                                />
+                                                {errors[f.name] && <span className="error">{errors[f.name]}</span>}
+                                            </div>
+
+                                            {isChildForm && isTitleField && mode === GenericFormMode.Edit && (
+                                                <div className={`title-actions`}>
+                                                    <button
+                                                        type="button"
+                                                        aria-label="edit-btn"
+                                                        className={`edit-btn ${mode}`}
+                                                        onClick={handleEditClick}
+                                                    />
+                                                    <button
+                                                        type="button"
+                                                        aria-label="delete-btn"
+                                                        className={`delete-btn delete-btn-icon ${isDeleting ? 'pressed' : ''}`}
+                                                        onClick={handleDeleteClick}
+                                                    />
+                                                </div>
+                                            )}
+                                        </div>
+                                    );
+                                })}
+
+                            {editable && (
+                                <div className="form-footer">
+                                    <div className="actions">
+                                        <Button type="button" onClick={handleCancel} buttonStyle="secondary">
+                                            {COMMON_TEXT_ADMIN.BUTTON.CANCEL}
+                                        </Button>
+                                        <Button
+                                            type="button"
+                                            onClick={
+                                                mode === GenericFormMode.Edit
+                                                    ? submit
+                                                    : () =>
+                                                          setModalConfig({
+                                                              title: DONATE_TEXT.QUESTION.BANK_DETAILS.ADD,
+                                                              onConfirm: submit,
+                                                          })
+                                            }
```

Comment:
Extract logic above to func `handleSubmitClick` please
```suggestion
                                            onClick={handleSubmitClick}